code/modules/jobs/job_types/detective.dm

```diff
@@ -65,6 +65,7 @@
 	gloves = /obj/item/clothing/gloves/color/black
 	head = /obj/item/clothing/head/fedora/det_hat
 	mask = /obj/item/cigarette
+	glasses = /obj/item/clothing/glasses/sunglasses/nuar
```

Comment:
I'd think it should be in the vendor as a premium item rather than make it equipped by default